File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/IChannelParsingState.vb
'*******************************************************************************************************
'  IChannelParsingState.vb - Channel parsing state interface - this is the parsing state root interface
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2005
'  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  02/18/2005 - J. Ritchie Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

''' <summary>This interface represents a protocol independent parsing state used by any kind of data.</summary>
''' <remarks>Data parsing is very format specific, classes implementing this interface create a common form for parsing state information particular to a data type.</remarks>
Public Interface IChannelParsingState

    ReadOnly Property DerivedType() As Type

    ReadOnly Property This() As IChannelParsingState

End Interface

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/IChannelParsingState.cs
//*******************************************************************************************************
//  IChannelParsingState.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  02/18/2005 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

namespace PCS.PhasorProtocols
{
    /// <summary>
    /// Represents a protocol independent interface representation of the parsing state used by any kind of
    /// <see cref="IChannel"/> data.<br/>
    /// This is the base interface implemented by all parsing state classes in the phasor protocols library;
    /// it is the root of the parsing state interface hierarchy.
    /// </summary>
    /// <remarks>
    /// Data parsing is very format specific, classes implementing this interface create a common form for
    /// parsing state information particular to a data type.
    /// </remarks>
    public interface IChannelParsingState
    {
        /// <summary>
        /// Gets or sets the length of the associated <see cref="IChannel"/> object being parsed from the binary image.
        /// </summary>
        int ParsedBinaryLength { get; set; }
    }
}